Add deployment manifests for k8s-gsm-tools

b01901143 opened the pull request:
Added deployment manifest for secret-rotator and secret-sync-controller.
Added rbac for k8s-gsm-tools svc account.

b01901143 committed:
Add deployment manifests for k8s-gsm-tools

Added deployment manifest for secret-rotator and secret-sync-controller.
Added rbac for k8s-gsm-tools svc account.

spiffxp commented:
/ok-to-test

spiffxp commented on config/prow/cluster/k8sgsmtools-rbac.yaml:
```diff
@@ -0,0 +1,33 @@
+kind: ServiceAccount
+metadata:
+  namespace: default
+  name: "ks8-gsm-tools"
```
`find . -name '*.yaml' -exec sed -i -e s/"k8s-gsm-tools"/k8s-gsm-tools/g` {} \;`
```suggestion
  name: ks8-gsm-tools
```

spiffxp replied:
Setup a workload identity annotation on this as if you had one, and I'll setup a GCP SA and binding etc. prior to merging this

eg: 
`iam.gke.io/gcp-service-account: k8s-gsm-tools@spiffxp-gke-dev.iam.gserviceaccount.com`

b01901143 replied:
Done.

I was following the pattern in `horologium-rbac.yaml` so that I added the double quotes on the name field.
Wonder why they're not working in this case.

spiffxp commented on config/prow/cluster/secretrotator-deployment.yaml:
```diff
@@ -0,0 +1,36 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: default
```
Can you add a `Namespace` resource to this PR and move everything into a `k8s-gsm-tools` namespace?

b01901143 replied:
I put both deployments and the service account into the `k8s-gsm-tools` namespace.
ClusterRole is not namespaced, so I left it as it was.

spiffxp commented on config/prow/cluster/secretrotator-deployment.yaml:
```diff
@@ -0,0 +1,36 @@
+      volumes:
+      - name: config-volume
+        configMap:
+          name: config
```
This ConfigMap is missing from the manifests here

You'll want it setup to rotate... uhh I know this isn't the exact file format, but hopefully this is enough for you to write it?
```
secret:
- namespace: test-pods
- name: service-account
- key: service-account.json
service-account:
- project: spiffxp-gke-dev
- name: spiffxp-gke-dev-prow
secretmanager:
- project: spiffxp-gke-dev
- secret: (whatever you want to call it)
```

spiffxp replied:
This secret should be the secret that allows prowjobs to write to GCS, and I have a job setup to run every 5min.  We'll be able to find out pretty quickly if rotation is working or not

b01901143 replied:
I pushed a configmap manifest that I think would work.
My question here was, if the job runs seamlessly, how are we going to distinguish whether the rotation is working fine or the service account key is simply not being rotated?

spiffxp reviewed (changes requested):
a few changes to make

b01901143 committed:
Add k8s-gsm-tools namespace

added manifest for namespace `k8s-gsm-tools`.
added manifest for configmap `config`.
put deployments into the `k8s-gsm-tools` namespace.

b01901143 commented on config/prow/cluster/k8sgsmtools-config.yaml:
```diff
@@ -0,0 +1,27 @@
+  rotConfig: |
+    specs:
+    - project: spiffxp-gke-dev
+      secret: service-account
```
For now, in order to bootstrap, the user needs to create at least an empty gsm secret `projects/spiffxp-gke-dev/secrets/service-account`. We can consider to upsert the gsm secret in the case that 1) the project exists, and 2) the secret does not yet exist.

spiffxp reviewed (approved):
/approve
/lgtm
